File: JavaScript/problem020.js
//Variables
factorial = BigInt(1);
let sum = 0;

//Finds 100 factorial
for(let i = 100; i > 1; i--){
    factorial *= BigInt(i);
}

//Converts it to string
factorial = factorial + "";

//Finds sum of all digits
for(let digit = 0; digit < factorial.length; digit++){
    sum += parseInt(factorial[digit]);
}

//Displays answer
console.log(sum); //Expected 648

File: Java/problem020.java
import java.math.BigInteger;

public class problem020 {
    public static void main(String[] args) {

        //Variables
        BigInteger factorial = new BigInteger("1");
        String factorialString;
        int sum = 0;

        //Finds 100 factorial
        for(int i = 100; i > 1; i--){
            factorial = factorial.multiply(BigInteger.valueOf(i));
        }

        //Converts it to string
        factorialString = factorial.toString();

        //Finds sum of all digits
        for(int digit = 0; digit < factorialString.length(); digit++){
            sum += Character.getNumericValue(factorialString.charAt(digit));
        }

        //Displays answer
        System.out.println(sum); //Expected 648
    }
}